src/registry/validation.ts

```diff
@@ -74,6 +74,46 @@ export const buildCommonSchema = (intl: IntlShape) =>
     key: buildKeySchema(intl),
   });
 
+// From https://zod.dev/?id=json-type
+const literalSchema = z.union([z.string(), z.number(), z.boolean(), z.null()]);
+type Literal = z.infer<typeof literalSchema>;
+type Json = Literal | {[key: string]: Json} | Json[];
+export const jsonSchema: z.ZodType<Json> = z.lazy(() =>
+  z.union([literalSchema, z.array(jsonSchema), z.record(jsonSchema)])
+);
+
+// Maps to @open-formulieren/types common.ts Option type.
+const optionTranslationSchema = z.object({
+  label: z.string(),
+});
+
+export const optionSchema = (intl: IntlShape) =>
+  z.object({
+    value: z
+      .string({
+        required_error: intl.formatMessage({
+          description: 'Form builder option value required error',
+          defaultMessage: 'The option value is a required field.',
+        }),
+      })
+      .min(1),
+    label: z
+      .string({
+        required_error: intl.formatMessage({
+          description: 'Form builder option label required error',
+          defaultMessage: 'The option label is a required field.',
+        }),
+      })
+      .min(1),
+    openForms: z
+      .object({
+        // zod doesn't seem to be able to use our supportedLanguageCodes for z.object keys,
+        // they need to be defined statically. So, 'record' it is.
+        translations: z.record(optionTranslationSchema.optional()),
```

Comment:
Does it work when specifying `z.record` keys as `z.union([z.literal('en'), z.literal('nl')])`? (Or built dynamically from `supportedLanguageCodes`, although you might need to ts ignore or use `as smth`)

Reply:
I tried building it dynamically and that doesn't work, and specifying them literally here defeats the purpose of having the `supportedLanguageCode`s in the builder context

Reply:
I've tried:

```ts
const supportedLanguageCodes = ['en', 'nl'];
z.record(z.union(supportedLanguageCodes.map(z.literal)), optionTranslationSchema.optional())
```

and it seems to work (but with a fixed value of `supportedLanguageCodes`, it's not coming from the context or anything). Am I missing something?

Reply:
the fixed value of `supportedLanguageCodes` is the problem - we centralize the list of supported language codes so that we don't have to update code in multiple places when we add additional languages.

And as you saw in the storybook build, that is set up with just a single language code, which causes `zod` to break because the `union` requires at least 2 elements in the union array.

Reply:
> And as you saw in the storybook build, that is set up with just a single language code, which causes `zod` to break because the `union` requires at least 2 elements in the union array.

This is what I was missing :+1: